### PR TITLE
Minor fixes and update of angular library versions

### PR DESCRIPTION
**Pull request by smeyn**
## Pull Request Template
Please, go through these steps before you submit a PR.

1. Make sure that your PR is not a duplicate.
2. **Before** submitting the PR, review your changes:
    * Adjust the code to the existing [style](https://google.github.io/styleguide).
    * Add a set of unit tests for the changes and ensure they all pass.
    * Add accompanying README.md file with instructions on usage. See [awesome-readme](https://github.com/matiassingers/awesome-readme) for good examples of high-quality READMEs.
    * Add a link to your contribution in the top-level [README](https://github.com/GoogleCloudPlatform/professional-services/blob/master/README.md) (alpha-order).
    * Add [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0) license headers with an up-to-date copyright date attributed to `Google LLC`.
    * Remove unnecessary LICENSE files. There's no need to include an additional license since all repository submissions are covered by the top-level Apache 2.0 [license](https://github.com/GoogleCloudPlatform/professional-services/blob/master/LICENSE).

3. **After** these steps, you're ready to open a pull request.
    * Give a descriptive title to your PR.
    * Provide a description of your changes.
    * Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).
<br/>

**PLEASE REMOVE THIS TEMPLATE BEFORE SUBMITTING**

### CODE REVIEW

File: tools/bq-visualizer/src/app/big-query.service.ts [hunk 1]
```diff
@@ -24,14 +24,19 @@ export class BigQueryService {
...
+  getQueryPlan(projectId: string, jobId: string, location: string):
+      Observable<Job> {
     // Extract job id if the job id is a collection of
     // [project]:[location].jobId
     const realid = jobId.split('.').slice(-1)[0];
     this.logSvc.debug(`getQueryPlan: fetched query plan for jobid=${jobId}`);
 
     const token = this.oauthService.getAccessToken();
-    const url = bqUrl(`/${projectId}/jobs/${realid}`, {access_token: token});
+    console.log(token);
```
Comment (dparrish) [inline tools/bq-visualizer/src/app/big-query.service.ts]: I think this debugging should be removed

File: tools/bq-visualizer/src/app/bq_query_plan.ts [hunk 2]
```diff
@@ -14,13 +14,27 @@ export class BqQueryPlan {
...
+    }
+    this.plan = plan;
     if (this.plan.statistics.query && this.plan.statistics.query.queryPlan) {
       this.nodes = this.plan.statistics.query.queryPlan;
     } else {
       logSvc.warn(`No query found in job ${this.plan.id}`);
+      return;
     }
+    this.isValid = true;
+    console.log(this);
```
Comment (dparrish) [inline tools/bq-visualizer/src/app/bq_query_plan.ts]: remove debugging?

File: tools/bq-visualizer/src/app/bq_query_plan.ts [hunk 3]
```diff
@@ -169,16 +194,59 @@ export class BqQueryPlan {
...
+      'endTime        ': new Date(endMs),
+      'start %        ': startPct.toLocaleString('en') + '% of job duration',
+      'end %          ': endPct.toLocaleString('en') + '% of job duration',
+    };
+    return JSON.stringify(result, null, 4);
+  }
   /** Return a formatted text of all details minus the steps. */
   getStageStats(node: QueryStage): string {
-    const result = {};
+    /*const result = {};
```
Comment (dparrish) [inline tools/bq-visualizer/src/app/bq_query_plan.ts]: Remove instead of commenting

File: tools/bq-visualizer/src/app/job/job.component.ts [hunk 4]
```diff
@@ -91,12 +91,19 @@ export class JobComponent implements OnDestroy {
...
   /** When selecting an item in the drop down list. */
   selectJob(job: BqJob): void {
-    this.bqService.getQueryPlan(job.projectId, job.id)
+    this.bqService.getQueryPlan(job.projectId, job.id, job.location)
         .pipe(takeUntil(this.destroy))
         .subscribe(
             detail => {
-              console.log('Got raw plan', detail);
-              this.planSelected.emit(new BqQueryPlan(detail, this.logSvc));
+              // console.log('Got raw plan', detail);
```
Comment (dparrish) [inline tools/bq-visualizer/src/app/job/job.component.ts]: remove

File: tools/bq-visualizer/src/app/rest_interfaces.ts [hunk 5]
```diff
@@ -65,16 +65,22 @@ interface Query {
...
   totalPartitionsProcessed: number;
   estimatedBytesProcessed: string;
+  reservationUsage: ReservationUsage[];
+  totalSlotMs: string;
   useQueryCache?: string;
   queryPlan?: QueryStage[];
   timeline: Timeline;
 }
-
+interface ReservationUsage {
```
Comment (dparrish) [inline tools/bq-visualizer/src/app/rest_interfaces.ts]: Add a blank line between top-level definitions

File: tools/bq-visualizer/src/app/timing-display/timing-display.component.ts [hunk 6]
```diff
@@ -29,14 +29,21 @@ export class TimingDisplayComponent implements OnInit {
...
     this.sideDisplay.stepDetails = [];
     this.sideDisplay.stageDetails = '';
     this.plan = plan;
   }
 
   async draw() {
-    if (!this.plan) return;
+    if (!this.plan) {
+      const gantt = document.getElementById('Gantt');
+      if ((gantt) && (gantt.childNodes.length > 0)) {
```
Comment (dparrish) [inline tools/bq-visualizer/src/app/timing-display/timing-display.component.ts]: Too many parentheses.

`if (gantt && gantt.childNodes.length > 0) {`

File: tools/bq-visualizer/src/tsconfig.app.json [hunk 7]
```diff
@@ -3,11 +3,14 @@
   "compilerOptions": {
     "outDir": "../out-tsc/app",
     "module": "es2015",
-    "types": ["node"],
-    "strictPropertyInitialization": true
+    "types": [
+      "node"
+    ],
+    "strictPropertyInitialization": false,
+    "strictNullChecks": false,
```
Comment (dparrish) [inline tools/bq-visualizer/src/tsconfig.app.json]: JSON doesn't allow a trailing comma on the last element of a map.

File: tools/bq-visualizer/src/tsconfig.spec.json [hunk 8]
```diff
@@ -7,7 +7,8 @@
       "jasmine",
       "node"
     ],
-    "strictPropertyInitialization": true
+    "strictPropertyInitialization": false,
+    "strictNullChecks": false,
```
Comment (dparrish) [inline tools/bq-visualizer/src/tsconfig.spec.json]: JSON doesn't allow a trailing comma on the last element of a map.

File: tools/bq-visualizer/src/app/big-query.service.ts (hunk 1 above)
Comment (smeyn) [reply tools/bq-visualizer/src/app/big-query.service.ts]: At this stage I still find situations where this is useful. So I am leaving itin until the code matures further.

File: tools/bq-visualizer/src/app/bq_query_plan.ts (hunk 2 above)
Comment (smeyn) [reply tools/bq-visualizer/src/app/bq_query_plan.ts]: done

File: tools/bq-visualizer/src/app/timing-display/timing-display.component.ts (hunk 6 above)
Comment (smeyn) [reply tools/bq-visualizer/src/app/timing-display/timing-display.component.ts]: done

**Comment by smeyn** [comment]
Addressed changes requested by David. 
In the mean time added minor bug fixes and added individual Aache 2 licence headers as laid out in pull request template.

File: tools/bq-visualizer/src/app/bq_query_plan.ts (hunk 3 above)
Comment (dparrish) [reply tools/bq-visualizer/src/app/bq_query_plan.ts]: This is still here as a commented-out block. Does it need to be here?

**Pull request merged**